src/node_report_module.cc

```diff
@@ -0,0 +1,315 @@
+// corresponding elements in the C++ world. Required because
+// (i) the tunables are first intercepted through the CLI but
+// later modified via APIs. (ii) the report generation events
+// are controlled partly from C++ and partly from JS.
+void SyncConfig(const FunctionCallbackInfo<Value>& info) {
+  Environment* env = Environment::GetCurrent(info);
+  Local<Context> context = env->context();
+  std::shared_ptr<PerIsolateOptions> options = env->isolate_data()->options();
+  IsolateData* id = env->isolate_data();
+
+  CHECK_EQ(info.Length(), 2);
+  Local<Object> obj = info[0]->ToObject(context).ToLocalChecked();
+  bool sync = info[1].As<Boolean>()->Value();
+
+  // events array
+  Local<String> eventskey = FIXED_ONE_BYTE_STRING(env->isolate(), "events");
+  Local<Value> events_unchecked =
+      obj->Get(context, eventskey).ToLocalChecked();
+  Local<Array> events;
+  if (events_unchecked->IsUndefined() || events_unchecked->IsNull()) {
+    events_unchecked = Array::New(env->isolate(), 0);
+    obj->Set(context, eventskey, events_unchecked);
+  }
+  events = events_unchecked.As<Array>();
+
+  // signal
+  Local<String> signalkey = FIXED_ONE_BYTE_STRING(env->isolate(), "signal");
+  Local<Value> signal_unchecked =
+      obj->Get(context, signalkey).ToLocalChecked();
+  Local<String> signal;
+  if (signal_unchecked->IsUndefined() || signal_unchecked->IsNull())
+    signal_unchecked = FIXED_ONE_BYTE_STRING(env->isolate(), "signal");
+  signal = signal_unchecked.As<String>();
+
+  Utf8Value signalstr(env->isolate(), signal);
+
+  // report file
+  Local<String> filekey = FIXED_ONE_BYTE_STRING(env->isolate(), "filename");
+  Local<Value> file_unchecked = obj->Get(context, filekey).ToLocalChecked();
+  Local<String> file;
+  if (file_unchecked->IsUndefined() || file_unchecked->IsNull())
+    file_unchecked = FIXED_ONE_BYTE_STRING(env->isolate(), "filename");
+  file = file_unchecked.As<String>();
+
+  Utf8Value filestr(env->isolate(), file);
+
+  // report file path
+  Local<String> pathkey = FIXED_ONE_BYTE_STRING(env->isolate(), "path");
+  Local<Value> path_unchecked = obj->Get(context, pathkey).ToLocalChecked();
+  Local<String> path;
+  if (path_unchecked->IsUndefined() || path_unchecked->IsNull())
+    path_unchecked = FIXED_ONE_BYTE_STRING(env->isolate(), "path");
+  path = path_unchecked.As<String>();
+
+  Utf8Value pathstr(env->isolate(), path);
+
+  // report verbosity
+  Local<String> verbosekey = FIXED_ONE_BYTE_STRING(env->isolate(), "verbose");
+  Local<Value> verbose_unchecked =
+      obj->Get(context, verbosekey).ToLocalChecked();
+  Local<Boolean> verbose;
+  if (verbose_unchecked->IsUndefined() || verbose_unchecked->IsNull())
+    verbose_unchecked = Boolean::New(env->isolate(), "verbose");
+  verbose = verbose_unchecked.As<Boolean>();
+
+  bool verb = verbose->BooleanValue(context).FromJust();
+
+  if (sync) {
+    std::string e = "exception";
+    std::string s = "signal";
+    std::string f = "fatalerror";
```

Comment:
Using `static const std::string` would avoid creating new string instances every time this code is hit

Reply:
done